internal/saucecloud/espresso.go

```diff
@@ -0,0 +1,103 @@
+package saucecloud
+
+import (
+	"fmt"
+	"github.com/rs/zerolog/log"
+	"github.com/saucelabs/saucectl/internal/espresso"
+	"github.com/saucelabs/saucectl/internal/job"
+	"strings"
+)
+
+// EspressoRunner represents the Sauce Labs cloud implementation for cypress.
+type EspressoRunner struct {
+	CloudRunner
+	Project espresso.Project
+}
+
+// RunProject runs the tests defined in cypress.Project.
+func (r *EspressoRunner) RunProject() (int, error) {
+	exitCode := 1
+
+	if err := r.validateTunnel(r.Project.Sauce.Tunnel.ID); err != nil {
+		return 1, err
+	}
+
+	appFileID, err := r.uploadProject(r.Project.Espresso.App, "application")
+	if err != nil {
+		return exitCode, err
+	}
+
+	testAppFileID, err := r.uploadProject(r.Project.Espresso.TestApp, "test application")
+	if err != nil {
+		return exitCode, err
+	}
+
+	passed := r.runSuites(appFileID, testAppFileID)
+	if passed {
+		exitCode = 0
+	}
+
+	return exitCode, nil
+}
+
+func (r *EspressoRunner) runSuites(appFileID string, testAppFileID string) bool {
+	sigChan := r.registerSkipSuitesOnSignal()
+	defer unregisterSignalCapture(sigChan)
+
+	jobOpts, results := r.createWorkerPool(r.Project.Sauce.Concurrency)
+	defer close(results)
+
+	// Submit suites to work on.
+	jobsCount := r.calculateJobsCount(r.Project.Suites)
+	go func() {
+		for _, s := range r.Project.Suites {
+			for _, d := range s.Devices {
+				for _, p := range d.PlatformVersions {
+					log.Debug().Str("suite", s.Name).Str("device", d.Name).Str("platform", p).Msg("Starting job")
+					jobsCount++
```

Comment:
This line is probably a leftover, since `jobsCount := r.calculateJobsCount(r.Project.Suites)` has the jobsCount already.